application/src/main/java/org/togetherjava/tjbot/commands/github/GitHubReference.java

```diff
@@ -0,0 +1,168 @@
+        }
+    }
+
+    /**
+     * Scans all repositories mentioned in the config and adds their object representations using
+     * kohuske's API to a list
+     */
+    private void acquireRepositories() throws IOException {
+        repositories = new ArrayList<>();
+
+        GitHub githubApi = GitHub.connectUsingOAuth(config.getGitHubApiKey());
+
+        for (long repo : config.getGitHubRepositories()) {
+            repositories.add(githubApi.getRepositoryById(repo));
+        }
+    }
+
+    @Override
+    public void onMessageReceived(MessageReceivedEvent event) {
+        if (event.getAuthor().isBot()) {
+            return;
+        }
+
+        Message message = event.getMessage();
+        String content = message.getContentRaw();
+        Matcher matcher = ISSUE_REFERENCE_PATTERN.matcher(content);
+        List<MessageEmbed> embeds = new ArrayList<>();
+
+        while (matcher.find()) {
+            int id = Integer.parseInt(matcher.group(1));
+
+            try {
+                Optional<GHIssue> issue = findIssue(id);
+
+                if (issue.isPresent()) {
+                    embeds.add(generateReply(issue.get()));
+                }
+            } catch (IOException ex) {
+                throw new UncheckedIOException(ex);
+            }
+        }
+
+        List<List<MessageEmbed>> partition = ListUtils.partition(embeds, 10);
+        boolean first = true;
+        TextChannel textChannel = message.getChannel().asTextChannel();
+
+        for (List<MessageEmbed> messageEmbeds : partition) {
+            if (first) {
+                message.replyEmbeds(messageEmbeds).mentionRepliedUser(false).queue();
+
+                first = false;
+            } else {
+                textChannel.sendMessageEmbeds(messageEmbeds).mentionRepliedUser(false).queue();
+            }
+        }
+    }
+
+    /**
+     * Generates the embed to reply with when someone references an issue
+     */
+    protected MessageEmbed generateReply(GHIssue issue) throws IOException {
+        return new EmbedBuilder()
+            .setColor(issue.getState() == GHIssueState.OPEN ? Color.green.getRGB()
+                    : Color.red.getRGB())
+            .setDescription(issue.getBody())
+            .setTitle("[#%d] %s".formatted(issue.getNumber(), issue.getTitle()),
+                    issue.getHtmlUrl().toString())
+            .setAuthor(issue.getUser().getName())
+            .setFooter("%s • %s".formatted(
+                    issue.getLabels()
+                        .stream()
+                        .map(GHLabel::getName)
+                        .collect(Collectors.joining(", ")),
+                    issue.getAssignees()
+                        .stream()
+                        .map(this::getUserNameOrFailAtRuntime)
+                        .collect(Collectors.joining(", "))))
+            .build();
+    }
+
+    // this is utterly stupid
+    /**
+     * Either properly gathers the name of a user or throws a UncheckedIOException
```

Comment:
what is this?

Reply:
the getName method of the api I'm using is declared with throws IOException which would clutter up the stream statement. this is basically just a helper method for a nicer looking stream

Reply:
fair